### PR TITLE
Add: `MultipleChoice with Transfer Learning` example

### PR DESCRIPTION
**Pull request by awsaf49**
This PR will add the MultipleChoice example with Transfer Learning using KerasNLP and KerasCore as mentioned in #791 .

Here is the colab demo: [Colab](https://colab.research.google.com/drive/18EhnA-912bBD-4mK81DFcUYh592ikCdm?usp=sharing)

I'm open to open to feedback to make this example more suitable for Keras users.

> I didn't run the `python autogen.py add_example vision/script_name` script as in the examples in this repo, I couldn't see any `/ipynb`, `/md`, or `/img` directories.

### CODE REVIEW

**Comment by awsaf49** [comment]
cc: @fchollet

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+!pip install -q keras-nlp --upgrade
+"""
+
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"  # or "tensorflow" or "torch"
+
+import keras_nlp
+import keras_core as keras
+import keras_core.backend as K
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:34]: Is this import used?

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+
+import jax
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:36]: Is this import used?

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+            "index": self.index,
+        }
+        return {**base_config, **config}
+
+
+def build_model():
+    # Define input layers
+    inputs = {
+        "token_ids": keras.Input(
+            shape=(4, None), dtype=tf.int32, name="token_ids"
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:496]: Use the string "int32"

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+        ),
+        "padding_mask": keras.Input(
+            shape=(4, None), dtype=tf.int32, name="padding_mask"
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:499]: Use the string "int32"

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+
+In this notebook, we'll experiment with an interesting augmentation technique,
+`OptionShuffle`. Since we're providing the model with one option at a time, we can
+introduce a shuffle to the order of options. For instance, options `[A, C, E, D, B]`
+would be rearranged as `[D, B, A, E, C]`. This practice will help the model focus on the
+content of the options themselves, rather than being influenced by their positions.
+"""
+
+
+def OptionShuffle(options, labels, prob=0.50, seed=None):
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:231]: Use snake case for functions

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+    if tf.random.uniform([]) > prob:  # Shuffle probability check
+        return options, labels
+    # Shuffle indices of options and labels in the same order
+    indices = tf.random.shuffle(tf.range(tf.shape(options)[0]), seed=seed)
+    # Shuffle options and labels
+    options = tf.gather(options, indices)
+    labels = tf.gather(labels, indices)
+    return options, labels
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:239]: This function is implementable with Keras Core backend agnostic APIs from `keras.ops`

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+
+keras.utils.set_random_seed(CFG.seed)
+
+"""
+## Accelerator
+Following codes automatically detects hardware (TPU or GPU).
+"""
+
+
+def init_device():
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:85]: Because examples are typically intended to run on Colab, we normally don't add distribution logic to them. In this case, the logic would make the example TF-only. But I think the whole example is pretty close to running with any backend.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Some intiail comments.

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 6 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:205]: I'm facing a bit of difficulty implementing it using only `keras.ops` as I couldn't find `keras.ops.shuffle` or `keras.ops.gather` functions. I was able to make it backend agnostic using `random` which shuffles regardless of the backend. But I'm not sure if it meets your expectations.

```py
import random

def option_shuffle(options, labels, prob=0.50, seed=None):
    if keras.random.uniform([]) > prob: 
        return options, labels
    indices = list(range(keras.ops.shape(options)[0]))
    indices = random.shuffle(indices)
    options = options[indices]
    labels = labels[indices]
    return options, labels
```

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 7 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: I've removed the distributed logic.

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 5 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: updated to snake case

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 4 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: updated to "int32"

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 2 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: No, thus I've removed it

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 1 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: No, thus I've removed it

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 6 above)
Comment (fchollet) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:205]: Your implementation is fine but it will only work in eager mode.

On second thought let's keep this function in pure TF since it is only used in the tf.data pipeline. We should add a comment explaining why we use TF here though.

Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:209]: I have added a note explaining the reasoning

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 8]
```diff
@@ -0,0 +1,571 @@
...
+provided along with a context and the model is trained to select the correct answer
+unlike question answering. We will use SWAG dataset to demonstrate this example.
+"""
+
+"""
+## Setup
+"""
+
+"""shell
+!pip install -q keras-core --upgrade
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:24]: Remove the `!` for shell statements, they're added automatically.

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py [hunk 9]
```diff
@@ -0,0 +1,571 @@
...
+"""
+## Augmentation
+
+In this notebook, we'll experiment with an interesting augmentation technique,
+`option_shuffle`. Since we're providing the model with one option at a time, we can
+introduce a shuffle to the order of options. For instance, options `[A, C, E, D, B]`
+would be rearranged as `[D, B, A, E, C]`. This practice will help the model focus on the
+content of the options themselves, rather than being influenced by their positions.
+
+> It is noteworthy that, even though `option_shuffle` function is written in pure
```
Comment (fchollet) [inline examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py:195]: Good comment! Please just format it as "Note:" rather than with `>`

Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: I've updated the comment

File: examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py (hunk 8 above)
Comment (awsaf49) [reply examples/keras_io/nlp/multiple_choice_task_with_transfer_learning.py]: Done

**Review by fchollet — APPROVED** [review]
LGTM -- thank you!
